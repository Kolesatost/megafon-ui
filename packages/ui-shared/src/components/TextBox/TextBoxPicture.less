@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-beta-text-box-picture');

@{b} {
    &_margin_default {
        margin: 16px 0;
    }

    &_margin_big_top_margin {
        margin: 24px 0 16px;
    }

    &_margin_big_bottom_margin {
        margin: 16px 0 24px;
    }

    &_margin_big_vertical_margin {
        margin: 24px 0;
    }

    &__img {
        width: 100%;
        height: 100%;
    }
}
