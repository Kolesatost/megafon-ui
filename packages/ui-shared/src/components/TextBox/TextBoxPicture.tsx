import * as React from 'react';
import { cnCreate } from '@megafon/ui-helpers';
import * as PropTypes from 'prop-types';
import './TextBoxPicture.less';

export const pictureMarginTypes = {
    DEFAULT: 'default',
    BIG_TOP: 'big-top',
    BIG_BOTTOM: 'big-bottom',
    BIG_VERTICAL: 'big-vertical',
} as const;

type PictureMarginTypesType = typeof pictureMarginTypes[keyof typeof pictureMarginTypes];

export interface ITextBoxPictureProps {
    /** URL картинки */
    url: string;
    /** Значение вертикальных отступов */
    margin?: PictureMarginTypesType;
}

const cn = cnCreate('mfui-beta-text-box-picture');
const TextBoxPicture: React.FC<ITextBoxPictureProps> = ({ url, margin = pictureMarginTypes.DEFAULT }) => (
    <div className={cn({ margin })}>
        <img className={cn('img')} src={url} alt="text-box" />
    </div>
);

TextBoxPicture.propTypes = {
    url: PropTypes.string.isRequired,
    margin: PropTypes.oneOf([
        pictureMarginTypes.DEFAULT,
        pictureMarginTypes.BIG_TOP,
        pictureMarginTypes.BIG_BOTTOM,
        pictureMarginTypes.BIG_VERTICAL,
    ]),
};

export default TextBoxPicture;
