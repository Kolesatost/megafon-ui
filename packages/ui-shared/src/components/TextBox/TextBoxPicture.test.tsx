import * as React from 'react';
import { shallow } from 'enzyme';
import TextBoxPicture, { ITextBoxPictureProps, pictureMarginTypes } from './TextBoxPicture';

const props: ITextBoxPictureProps = {
    url: '#',
};

describe('TextBoxPicture', () => {
    it('render with default props', () => {
        const wrapper = shallow(
            <TextBoxPicture {...props} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render with big vertical margins', () => {
        const wrapper = shallow(
            <TextBoxPicture {...props} margin={pictureMarginTypes.BIG_VERTICAL_MARGIN} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render with big bottom margin', () => {
        const wrapper = shallow(
            <TextBoxPicture {...props} margin={pictureMarginTypes.BIG_BOTTOM_MARGIN} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render with big top margin', () => {
        const wrapper = shallow(
            <TextBoxPicture {...props} margin={pictureMarginTypes.BIG_TOP_MARGIN} />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
