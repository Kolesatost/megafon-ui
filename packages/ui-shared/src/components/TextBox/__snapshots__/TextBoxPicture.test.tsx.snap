// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBoxPicture render with big bottom margin 1`] = `
<div
  className="mfui-beta-text-box-picture mfui-beta-text-box-picture_margin_big-bottom"
>
  <img
    alt="text-box"
    className="mfui-beta-text-box-picture__img"
    src="#"
  />
</div>
`;

exports[`TextBoxPicture render with big top margin 1`] = `
<div
  className="mfui-beta-text-box-picture mfui-beta-text-box-picture_margin_big-top"
>
  <img
    alt="text-box"
    className="mfui-beta-text-box-picture__img"
    src="#"
  />
</div>
`;

exports[`TextBoxPicture render with big vertical margins 1`] = `
<div
  className="mfui-beta-text-box-picture mfui-beta-text-box-picture_margin_big-vertical"
>
  <img
    alt="text-box"
    className="mfui-beta-text-box-picture__img"
    src="#"
  />
</div>
`;

exports[`TextBoxPicture render with default props 1`] = `
<div
  className="mfui-beta-text-box-picture mfui-beta-text-box-picture_margin_default"
>
  <img
    alt="text-box"
    className="mfui-beta-text-box-picture__img"
    src="#"
  />
</div>
`;
